### PR TITLE
Improve tests and code coverage

### PR DESCRIPTION
**Pull request by gregurco**
- [ ] Bug fix
- [ ] New feature
- [ ] BC breaks
- [ ] Deprecations
- [x] Tests pass

Fixed tickets: []
License: MIT

Todo:
- [x] tests for plugins system
- [x] review issues from scrutinizer

### CODE REVIEW

File: src/DataCollector/HttpDataCollector.php [hunk 1]
```diff
@@ -160,24 +160,10 @@ protected function getLogGroup(string $id) : LogGroup
      */
     public final function getIconColor() : string
     {
-        if ((float)$this->getSymfonyVersion() >= 2.8) {
+        if (version_compare(Kernel::VERSION, '2.8.0', '>=')) {
```
Comment (gregurco) [inline src/DataCollector/HttpDataCollector.php]: @florianpreusner I replaced custom function by php function
Doc: http://php.net/manual/en/function.version-compare.php

File: src/DependencyInjection/Compiler/EventHandlerCompilerPass.php [hunk 2]
```diff
@@ -27,9 +28,7 @@ public function process(ContainerBuilder $container)
 
         foreach ($taggedServices as $id => $tags) {
             foreach ($tags as $attributes) {
-                if (array_key_exists('service', $attributes)
-                    && false !== strstr($attributes['event'], 'guzzle_bundle')) {
-
+                if (isset($attributes['service']) && in_array($attributes['event'], GuzzleEvents::EVENTS, true)) {
```
Comment (gregurco) [inline src/DependencyInjection/Compiler/EventHandlerCompilerPass.php]: @florianpreusner here we had bug. When we migrated to flex the events were renamed. For example  `guzzle_bundle.pre_transaction` became `eight_points_guzzle.pre_transaction` but this verification was not adapted. I fixed it and added more strict verification.

**Comment by gregurco** [comment]
@florianpreusner coverage result:
```
OK (100 tests, 258 assertions)

Summary:                  
    Classes: 81.25% (13/16)  
    Methods: 97.52% (118/121)
    Lines:   97.71% (469/480)
```

Now it looks better :slightly_smiling_face:

**Comment by gregurco** [comment]
@florianpreusner coverage result:

OK (102 tests, 262 assertions)

Summary:                  
    Classes: 93.75% (15/16)  
    Methods: 99.17% (120/121)
    Lines:   99.79% (479/480)

Now it looks better :slightly_smiling_face:

File: src/DataCollector/HttpDataCollector.php (hunk 1 above)
Comment (florianpreusner) [reply src/DataCollector/HttpDataCollector.php]: 👍  nice!

File: src/DependencyInjection/Compiler/EventHandlerCompilerPass.php (hunk 2 above)
Comment (florianpreusner) [reply src/DependencyInjection/Compiler/EventHandlerCompilerPass.php]: good shot! 👍

**Comment by florianpreusner** [comment]
It looks much better!

**Comment by florianpreusner** [comment]
Released in v7.3.0 (https://github.com/8p/EightPointsGuzzleBundle/releases/tag/7.3.0)

**Pull request merged**